### PR TITLE
Fix `nn.functional` Benchmark Fallback Labels

### PR DESCRIPTION
**Pull request by peterdsharpe**
## Problem

The functional benchmark plotter reconstructed unlabeled ASV result positions in
a different order than the ASV benchmark runner produced them.

- The runner ordered benchmark keys as `spec -> phase -> impl -> case`.
- The plot fallback ordered keys as `phase -> spec -> impl -> case`.

When ASV result payloads did not include explicit labels, the plotter assigned
timing values to the wrong benchmark keys. This could make plots attribute one
measurement to a different functional or phase, such as showing a backward KNN
timing under a RadiusSearch forward case.

There was also a related fallback drift: the plotter derived fallback labels
from plottable specs only, while the runner times all benchmark specs that have
available implementations. That can shift every later unlabeled value whenever a
single-implementation spec appears in the ASV vector.

## Changes

- Added `BenchmarkKey` and `build_benchmark_plan()` to
  `benchmarks/physicsnemo/nn/functional/_spec_utils.py` so the benchmark runner
  and plot fallback share the same ASV key construction logic.
- Updated `benchmark_functionals.py` to use the shared plan builder for
  `_BENCHMARK_KEYS` and `_KEY_TO_SPEC`.
- Updated `plot_functional_benchmarks.py` so `_build_fallback_params()` rebuilds
  the full ASV key vector from the registry, then lets the existing grouping
  logic filter to plottable specs.
- Added a type annotation to `FUNCTIONAL_SPECS` so the shared helper remains
  type-checkable.
- Added regression tests covering both the exact fallback key order and the
  unlabeled value-to-key mapping, including a single-implementation spec that is
  timed by ASV but not plotted.

## Validation

```text
uv run --no-sync pytest test/nn/functional/test_functional_benchmarks.py -q

2 passed
```

`ReadLints` reported no diagnostics for the edited files, and `git diff --check`
passed.

### CODE REVIEW

File: benchmarks/physicsnemo/nn/functional/_spec_utils.py [hunk 1]
```diff
@@ -128,9 +133,42 @@ def case_by_index(
     )
 
 
+def build_benchmark_plan(
+    *,
+    device: torch.device | str,
+    phases: Iterable[str],
+    selected_specs: Iterable[type[FunctionSpec]],
+) -> tuple[list[BenchmarkKey], dict[BenchmarkKey, type[FunctionSpec]]]:
```
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply benchmarks/physicsnemo/nn/functional/_spec_utils.py:141]: Fixed

**Review by loliverhennigh — APPROVED** [review]
LGTM
